vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -158,32 +213,33 @@ def pre_build(self):
                 )
             ]
 
-    def __call__(self, target_to_data_frame: dict[ModelID, pd.DataFrame]):
-        # Only relevant for a dynamic filter.
-        # Although targets are fixed at build time, the validation logic is repeated during runtime, so if a column
-        # is missing then it will raise an error. We could change this if we wanted.
-        # Call this from actions_utils
-        targeted_data = self._validate_targeted_data(
-            {target: data_frame for target, data_frame in target_to_data_frame.items() if target in self.targets},
-            eagerly_raise_column_not_found_error=True,
-        )
-
-        if (column_type := self._validate_column_type(targeted_data)) != self._column_type:
-            raise ValueError(
-                f"{self.column} has changed type from {self._column_type} to {column_type}. A filtered column cannot "
-                "change type while the dashboard is running."
-            )
-
-        # TODO: when implement dynamic, will need to do something with this e.g. pass to selector.__call__.
-        # if isinstance(self.selector, SELECTORS["numerical"] + SELECTORS["temporal"]):
-        #     options = self._get_options(targeted_data)
-        # else:
-        #     # Categorical selector.
-        #     _min, _max = self._get_min_max(targeted_data)
-
     @_log_call
     def build(self):
-        return self.selector.build()
+        selector_build_obj = self.selector.build()
+        # TODO: Align the (dynamic) object's return structure with the figure's components when the Dash bug is fixed.
+        #  This means returning an empty "html.Div(id=self.id, className=...)" as a placeholder from Filter.build().
+        #  Also, make selector.title visible when the filter is reloading.
+        if not self._dynamic:
+            return selector_build_obj
+
+        # Temporarily hide the selector and numeric dcc.Input components during the filter reloading process.
+        # Other components, such as the title, remain visible because of the configuration:
+        # overlay_style={"visibility": "visible"} in dcc.Loading.
+        # Note: dcc.Slider and dcc.RangeSlider do not support the "style" property directly,
+        # so the "className" attribute is used to apply custom CSS for visibility control.
+        # Reference for Dash class names: https://dashcheatsheet.pythonanywhere.com/
+        selector_build_obj[self.selector.id].className = "invisible"
+        if f"{self.selector.id}_start_value" in selector_build_obj:
+            selector_build_obj[f"{self.selector.id}_start_value"].className = "d-none"
+        if f"{self.selector.id}_end_value" in selector_build_obj:
+            selector_build_obj[f"{self.selector.id}_end_value"].className = "d-none"
+
+        return dcc.Loading(
```

Comment:
New dynamic filter UI of during the reloading process can be tested by adding `time.sleep(3)` within `_get_modified_page_figures` function in the `_actions_utils.py` file.

Reply:
Nice, this works pretty well for now I think! Clever solution.

The reason I say "for now" is that I don't think it's going to work well for the case that you update a filter on the page without calling `page.build` i.e. in future when you can target a filter and update it with a parameter. 

For now we can leave it like this and think about it again when we enable that.